### PR TITLE
Add matmul_2d benchmark test 

### PR DESCRIPTION
**Pull request by yugaoTT**
### Problem description
Add the benchmark for matmul 2d, with some best (or almost) configs for reference.

### Checklist
- [ ] Post commit CI https://github.com/tenstorrent/tt-metal/actions/runs/11612449870

### CODE REVIEW

File: tests/ttnn/unit_tests/benchmarks/test_benchmark.py [hunk 1]
```diff
@@ -66,3 +71,217 @@ def test_benchmark_ttnn_matmul(device, use_program_cache, m_size, k_size, n_size
...
+                ttnn.execute_trace(device, tid, cq_id=0, blocking=False)
+
+            profiler.start(f"run")
+            for iter in range(0, num_measurement_iterations):
+                ttnn.execute_trace(device, tid, cq_id=0, blocking=False)
+            ttnn.synchronize_device(device)
+            profiler.end(f"run")
+            ttnn.DumpDeviceProfiler(device)
+            inference_time_avg = profiler.get("run") / num_measurement_iterations
+            tflops = 2 * m * k * n / 1e12 / inference_time_avg
```
Comment (TT-BrianLiu) [inline tests/ttnn/unit_tests/benchmarks/test_benchmark.py:280]: This is assuming WH clock speed? I would put these into sensible constants

**Review by TT-BrianLiu — APPROVED** [review]
Minor comment

File: tests/ttnn/unit_tests/benchmarks/test_benchmark.py (hunk 1 above)
Comment (yugaoTT) [reply tests/ttnn/unit_tests/benchmarks/test_benchmark.py:280]: no clk speed assumed, 1e12 is to get tflops from flops

Comment (fleclairTT) [reply tests/ttnn/unit_tests/benchmarks/test_benchmark.py:280]: Correct, and the factor of 2 is because we get 2 operations from an FMA

**Pull request merged**